mathematical-companion/main.tex

```diff
@@ -0,0 +1,1300 @@
+``backsolve'' by subtracting (i.e.~adding) all the rows above it. This will place
+the residue, our desired checksum, in the correct place.
+
+\section{Secret Sharing}
+
+In the Mathematical Preliminaries section we described Lagrange interpolation and
+Shamir's Secret Sharing Scheme (SSSS). We defined the Fundamental Theorem of Computing
+SSSS With Volvelles as the observation that Lagrange interpolation allows evaluating
+a polynomial at a fixed point as an affine combination of its evaluations at other
+fixed points.
+
+As a consequence, any linear or affine relationship between the original evalutations
+will be preserved. We will come to this in a moment, but first let's describe the SSSS
+process.
+
+\subsection{Computing SSSS}
+
+There are two places where we use Lagrange interpolation:
+\begin{itemize}
+\item When creating a share with index $x$, we start with $k$ initial shares $x_1$,
+$x_2$, \ldots, $x_k$ whose indices are always the first $k$ symbols of the bech32
+alphabet \vc{A}, \vc{C}, \vc{D}, etc.
+
+(There is an alternate process where the user starts with a fixed secret, in which
+case $x_1$ is \vc{S} and the other $x_i$'s are shifted, but the rest is exactly
+the same.)
+\item When recovering a secret, we start with $k$ shares with indices $\{x_i\}_{i=1}^k$
+which are fixed during recovery but unpredictable in advance, and compute the
+\vc{S} share.
+\end{itemize}
+
+In both cases, the computation is straightforward: evaluate~\eqref{eq:linterp} as
+\[ p(x) = \sum_{i=1}^k \ell_i(x) y_i \]
+Here $y_i$ is the data of the $i$th share and $\ell_i(x)$ is a Lagrange multiplier
+which must be computed by the user. We refer to these multiplies as ``Recovery
+Symbols'' in the text. To avoid confusion, they are always encoded using the symbol
+alphabet rather than bech32. The process for obtaining these is:
+
+\begin{itemize}
+\item During share creation, the evaluation points $\{x_i\}$, which are the
+user's initial share indices, are always fixed. Therefore $\ell_i(x)$ is purely
+a function of $x$ (the index of the share to be created) and $k$. There aren't
+that many possibilities so we simply provide tables on the ``Constructing Shares''
+page.
+
+(It is instructive to modify the PostScript source for these tables to allow
+``deriving'' the initial shares from themselves. You will find that the index for the initial
+share under question becomes 1 ($\aleph$) while the index for all the other shares
+is 0 ($\times$).)
+
+\item During recovery, there are up to 31 outstanding non-\vc{S} shares and
+the user has an arbitrary subset of $k$ of them. There are $\binom{31}k$
+possibilities, which for $k=2$ is a reasonable number --- 465 --- but for
+$k\geq3$ quickly grows out of hand, to 4495, then 31465, and beyond.
+
+For $k = 2$ we provide a table, both in tabular form and in the form of a volvelle.
+This is what the Recovery Wheel ultimately is, and at position $p$, window $w$ it
+computes
+\[ \ell_i(\vc{S}) = \frac{w + \vc{S}}{w + p} \]
```

Comment:
`\ell_i(w)`

Reply:
I think it should be `ell_i(S)`. The idea is that `\ell_i(x)` is a polynomial in an indeterminate `x` ... and we are evaluating it at `S` when doing recovery.